test/pylib/minio_server.py

```diff
@@ -223,6 +227,12 @@ async def main():
         print(f'export S3_SERVER_ADDRESS_FOR_TEST={server.address}')
         print(f'export S3_SERVER_PORT_FOR_TEST={server.port}')
         print(f'export S3_PUBLIC_BUCKET_FOR_TEST={server.bucket_name}')
+        if args.output_config:
+            with open(args.output_config, 'w', encoding='ascii') as config_file:
```

Comment:
I think it makes sens to generate this config unconditionally. And patch the `test/object_store/test_basic.py` to pick it up automagically, instead of generating it on its own

Reply:
> how about exporting yet another environmental variable like S3_SERVER_CONFIG, so the underlying tests can use it if they want ?

Ah, minio server generates this config in "random" tempdir? In that case yes, env variable would work.
OTOH, maybe putting it right into testlog dir would work too.

Reply:
we can do both

1. add a new method to `MinioServer` to generate the conf file
2. put it in the specified dir
3. `test.py` will generate the conf file in the testlog dir, and remove it when stopping minio server. but because we don't have a dedicate channel allowing the tests launched by `test.py` query information about the fixtures from `test.py`,  we've been using env variables for passing the settings. `TMPDIR` is just one of them. we could hardwire the filename in the `test_basic.py`, but i would avoid doing so.
4. `minio_server.py` will generate the conf file at the specified path, so developer can use it to perform manual tests